### PR TITLE
Fixed all Errors and Building for arm Arch

### PR DESCRIPTION
**Pull request by l0nax**

### CODE REVIEW

File: .ci/run.sh [hunk 1]
```diff
@@ -10,4 +10,8 @@ if [[ "$(uname -s)" == 'Darwin' ]]; then
     pyenv activate conan
 fi
 
+## create Environment Variables
```
Comment (SSE4) [inline .ci/run.sh:13]: this is not necessary, environment variable is automatically set by bincrafters-package-tools

File: .travis.yml [hunk 2]
```diff
@@ -28,6 +25,21 @@ matrix:
         env: CONAN_CLANG_VERSIONS=5.0 CONAN_DOCKER_IMAGE=lasote/conanclang50
```
Comment (SSE4) [inline .travis.yml]: please change lasote's images to conanio

File: build.py [hunk 3]
```diff
@@ -2,9 +2,15 @@
 # -*- coding: utf-8 -*-
 
 from cpt.packager import ConanMultiPackager
+import os
 
 if __name__ == "__main__":
-	builder = ConanMultiPackager(username='bincrafters');
-	
+	builder = ConanMultiPackager(username='bincrafters',
```
Comment (SSE4) [inline build.py:8]: this also seems to be redundant

File: conanfile.py [hunk 4]
```diff
@@ -9,7 +9,7 @@ class libsocket(ConanFile):
     name = "libsocket"
     version = "2.4.1"
     description = "Conan.io Package for libsocket Libary."
-    url = "https://github.com/l0nax/conan-libsocket"
+    url = "https://github.com/bintray/conan-libsocket"
```
Comment (SSE4) [inline conanfile.py]: should be `bincrafters`

File: conanfile.py [hunk 5]
```diff
@@ -19,7 +19,8 @@ class libsocket(ConanFile):
     exports_sources = ["CMakeLists.txt"]
     generators = "cmake"
 
-    settings = {"os": ["Linux"], "arch": None, "compiler": None, "build_type": None}
+    #settings = {"os": ["Linux"], "arch": None, "compiler": None, "build_type": None}
```
Comment (SSE4) [inline conanfile.py]: remove commented code

File: test_package/conanfile.py [hunk 6]
```diff
@@ -16,5 +15,17 @@ def build(self):
...
+        p = subprocess.Popen(["file", os.path.join("bin", "test_package")], stdout=subprocess.PIPE, shell=True)
+        output = p.communicate()[0]
+        
+        arch = platform.architecture()
+        
+        ## only Execute if it's possible
+        ## it's not releay a nice way..
+        if arch == "64bit":
+            if "64-bit" in output and "x86-64" in output: 
+                self.run("LD_LIBRARY_PATH=%s %s"%(os.environ.get('LD_LIBRARY_PATH', ''),os.path.join("bin","test_package")))
```
Comment (SSE4) [inline test_package/conanfile.py]: use `self.run(exe, run_environment=True)`

File: conanfile.py [hunk 7]
```diff
@@ -32,9 +33,14 @@ class libsocket(ConanFile):
     source_subfolder = "source_subfolder"
     build_subfolder = "build_subfolder"
 
+
     def source(self):
         self.run("git clone https://github.com/dermesser/libsocket --depth 1 %s" % self.source_subfolder)
 
+        # patching CMakeLists.txt in Source Directory
```
Comment (SSE4) [inline conanfile.py]: are patches really applied?
if not, please remove commented code and patch file

**Review by SSE4 — CHANGES_REQUESTED** [review]

File: .ci/run.sh (hunk 1 above)
Comment (l0nax) [reply .ci/run.sh:13]: I've tried to run it without exporting the Variable.
But it doesn't worked.

File: build.py (hunk 3 above)
Comment (l0nax) [reply build.py:8]: No, it does not run without the `username` Parameter

**Review by l0nax — COMMENTED** [review]
Changed Code

File: build.py (hunk 3 above)
Comment (SSE4) [reply build.py:8]: weird, no parameters and environment variables should be necessary at all
it works for all recipes with no issues
I'll check CI settings and let you know what's going wrong

**Pull request closed without merging**